### PR TITLE
Refactor font.render

### PR DESCRIPTION
**Pull request by Starbuck5**
Why?
Well, I kinda just started to do it... spent some time on it... then finished doing it.

It gets rid of some bit rot, simplifies the code a bit, adds a couple new comments. I hope this makes the font.render function a better place for future expansions with recent SDL_ttf features.

### CODE REVIEW

**Comment by Starbuck5** [comment]
This was intended as purely a code quality thing, but benchmarking shows this happens to give a 4% speedup too.

File: src_c/font.c [hunk 1]
```diff
@@ -371,15 +371,14 @@ static PyObject *
...
-    int aa;
+    int antialias;
     PyObject *text, *final;
-    PyObject *fg_rgba_obj, *bg_rgba_obj = NULL;
+    PyObject *fg_rgba_obj, *bg_rgba_obj = Py_None;
     Uint8 rgba[] = {0, 0, 0, 0};
     SDL_Surface *surf;
-    SDL_Color foreg, backg;
-    int just_return;
+    const char *astring = "\x00";
```
[the remark this answers is not in the document]
Comment (Starbuck5) [reply src_c/font.c]: You're right. I thought it wouldn't work, as I've had weird experiences with 0 length arrays in C. But I guess this ends up as a 1 length array.

**Comment by Starbuck5** [comment]
I made that change you suggested Ankith, then I squashed and rebased to make things tidy.

File: src_c/font.c [hunk 2]
```diff
@@ -388,121 +387,83 @@ font_render(PyObject *self, PyObject *args)
...
-        astring = PyBytes_AsString(bytes);
-        if (strlen(astring) != (size_t)PyBytes_GET_SIZE(bytes)) {
-            Py_DECREF(bytes);
+        if (strlen(astring) != (size_t)_size) {
             return RAISE(PyExc_ValueError,
                          "A null character was found in the text");
         }
+    }
+
+    if (PyBytes_Check(text)) {
```
Comment (ankith26) [inline src_c/font.c:419]:
```suggestion
    else if (PyBytes_Check(text)) {
```
We don't need to bother with a bytes check if `text` was already unicode

**Review by ankith26 — APPROVED** [review]
Nice cleanups, the code is sure simpler now! 🥳 🎈 

Left a minor nitpick review but approved nonetheless, resolve at will

File: src_c/font.c [hunk 3]
```diff
@@ -371,15 +371,14 @@ static PyObject *
...
-    PyObject *fg_rgba_obj, *bg_rgba_obj = NULL;
+    PyObject *fg_rgba_obj, *bg_rgba_obj = Py_None;
     Uint8 rgba[] = {0, 0, 0, 0};
     SDL_Surface *surf;
-    SDL_Color foreg, backg;
-    int just_return;
+    const char *astring = "";
 
-    if (!PyArg_ParseTuple(args, "OpO|O", &text, &aa, &fg_rgba_obj,
+    if (!PyArg_ParseTuple(args, "OpO|O", &text, &antialias, &fg_rgba_obj,
```
Comment (MyreMylar) [inline src_c/font.c:381]: Could take this opportunity to add keyword arguments if you fancy?

Or just leave it for another PR.

**Review by MyreMylar — APPROVED** [review]
LGTM 👍  tested it locally with local test run and rendering text with a variety of options in a test program.

File: src_c/font.c [hunk 4]
```diff
@@ -388,121 +387,83 @@ font_render(PyObject *self, PyObject *args)
...
-    foreg.b = rgba[2];
-    foreg.a = SDL_ALPHA_OPAQUE;
-    if (bg_rgba_obj == Py_None) {
-        /* Explicit None is the same as not passing a color for us */
-        bg_rgba_obj = NULL;
-    }
-    if (bg_rgba_obj != NULL) {
+
+    SDL_Color foreg = {rgba[0], rgba[1], rgba[2], SDL_ALPHA_OPAQUE};
+    /* might be overriden right below, with an explicit background color */
```
Comment (novialriptide) [inline src_c/font.c:392]: "overriden" should be "overridden"

**Review by novialriptide — CHANGES_REQUESTED** [review]
I haven't tested it but I just found a spelling mistake.

File: src_c/font.c (hunk 4 above)
Comment (Starbuck5) [reply src_c/font.c:392]: Thanks.

File: src_c/font.c (hunk 3 above)
Comment (Starbuck5) [reply src_c/font.c:381]: I'll leave it for another PR. I did an experiment with METH_FASTCALL on this, and got an additional 4% speedup for rendering text. But I'm not trying to merge that because FASTCALL is 3.7 and up, and there isn't a good way to parse keywords with it.
